ssl/t1_lib.c

```diff
@@ -172,6 +172,15 @@ static const TLS_GROUP_INFO nid_list[] = {
     {EVP_PKEY_X25519, 128, TLS_GROUP_CURVE_CUSTOM, 0x001D}, /* X25519 (29) */
     {EVP_PKEY_X448, 224, TLS_GROUP_CURVE_CUSTOM, 0x001E}, /* X448 (30) */
 # endif /* OPENSSL_NO_EC */
+#ifndef OPENSSL_NO_GOST
+    {NID_id_tc26_gost_3410_2012_256_paramSetA, 128, TLS_GROUP_CURVE_PRIME, 0x0022}, /* GC256A (34) */
+    {NID_id_tc26_gost_3410_2012_256_paramSetB, 128, TLS_GROUP_CURVE_PRIME, 0x0023}, /* GC256B (35) */
+    {NID_id_tc26_gost_3410_2012_256_paramSetC, 128, TLS_GROUP_CURVE_PRIME, 0x0024}, /* GC256C (36) */
+    {NID_id_tc26_gost_3410_2012_256_paramSetD, 128, TLS_GROUP_CURVE_PRIME, 0x0025}, /* GC256D (37) */
+    {NID_id_tc26_gost_3410_2012_512_paramSetA, 256, TLS_GROUP_CURVE_PRIME, 0x0026}, /* GC512A (38) */
+    {NID_id_tc26_gost_3410_2012_512_paramSetB, 256, TLS_GROUP_CURVE_PRIME, 0x0027}, /* GC512B (39) */
+    {NID_id_tc26_gost_3410_2012_512_paramSetC, 256, TLS_GROUP_CURVE_PRIME, 0x0028}, /* GC512C (40) */
+#endif
```

Comment:
One more question I want to raise.
Formally GOST groups can be treated as TLS_GROUP_CURVE_PRIME, but as GOST is not (always) implemented using openssl EC library, it causes build failure for the no-ec Travis build.

It makes sense provide a separate flag like TLS_GROUP_CURVE_GOST for these curves.

Reply:
No, introducing TLS_GROUP_CURVE_GOST does not do the trick. I've added no-gost into the test script build options.